src/app/mod.rs

```diff
@@ -23,204 +26,412 @@ pub fn setup() {
+	/// const person = await db.select('person:h5wxrf2ewk8xjxosxtyc');
+	/// ```
+	pub async fn select(&self, resource: String) -> Result<JsValue, Error> {
+		let resource = Resource::from(resource);
+		let response = self.db.select(resource).await?;
+		Ok(to_value(&Json::from(response))?)
 	}
 
-	#[wasm_bindgen]
-	pub fn create(&self, what: JsValue, data: JsValue) -> Promise {
-		match &self.kind {
-			Kind::None => Promise::reject(&JsValue::from(Error::MustInstantiate)),
-			Kind::Local(v) => future_to_promise(Local::create(v.clone(), what, data)),
-			Kind::Remote(_) => todo!(),
-		}
+	/// Create a record in the database
+	///
+	/// ```js
+	/// // Create a record with no fields set
+	/// const person = await db.create('person');
+	///
+	/// Create a record with fields set
+	/// const record = {
+	///     name: 'Tobie',
+	///     settings: {
+	///         active: true,
+	///         marketing: true
+	///     }
+	/// };
+	/// const person = db.create('person', { content: record });
+	/// ```
+	pub async fn create(&self, resource: String, data: JsValue) -> Result<JsValue, Error> {
+		let resource = Resource::from(resource);
+		let response = match from_value::<Option<opt::resource::Create>>(data)? {
+			Some(data) => self.db.create(resource).content(data.content).await?,
+			None => self.db.create(resource).await?,
+		};
+		Ok(to_value(&Json::from(response))?)
 	}
 
-	#[wasm_bindgen]
-	pub fn update(&self, what: JsValue, data: JsValue) -> Promise {
-		match &self.kind {
-			Kind::None => Promise::reject(&JsValue::from(Error::MustInstantiate)),
-			Kind::Local(v) => future_to_promise(Local::update(v.clone(), what, data)),
-			Kind::Remote(_) => todo!(),
-		}
+	/// Update all records in a table, or a specific record
+	///
+	/// ```js
+	/// const record = {
+	///     name: 'Tobie',
+	///     settings: {
+	///         active: true,
+	///         marketing: true
+	///     }
+	/// };
+	///
+	/// // Replace the current document / record data with the specified data.
+	/// const person = db.update('person:tobie', { content: record });
+	///
+	/// // Merge the current document / record data with the specified data.
+	/// const person = db.update('person:tobie', { merge: record });
+	///
+	/// // Apply JSON Patch changes to all records, or a specific record, in the database.
+	/// const person = db.update('person:tobie', {
+	///     patch: { op: 'replace', path: '/settings/active', value: false }
+	/// });
+	/// ```
+	pub async fn update(&self, resource: String, data: JsValue) -> Result<JsValue, Error> {
+		let resource = Resource::from(resource);
+		let Some(data) = from_value::<Option<opt::resource::Update>>(data)? else {
+            let response = self.db.update(resource).await?;
+            return Ok(to_value(&Json::from(response))?);
+        };
+		let response = match (data.content, data.merge, data.patch) {
+			(None, None, None) => {
+				return Err("Expected content, merge or patch in the update method".into());
+			}
```

Comment:
A user should be able to run `UPDATE` on a resource (table, thing) without needing to apply a `CONTENT`, `MERGE`, or `PATCH` clause.

Reply:
This is already possible. When none of those clauses are supplied we short-circuit the response. By the time we get to this, `db.update(resource)` would have returned already.

This message is simply there to stop users from entering an empty object (`await db.update(resource, {})`). In hindsight it might be better to use an enum here instead. I will change it to that.

Reply:
I updated the struct and moved the clauses into an enum.